plugins/modules/database/misc/redis_incr.py

```diff
@@ -0,0 +1,164 @@
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: redis_incr
+short_description: Increment keys in Redis
+version_added: 3.7.0
+description:
+   - Increment integers or float keys in Redis database and get new value.
+   - Default increment for all keys is 1. For specific increments use the
+     C(increment_int) and C(increment_float) options.
+author: "Andreas Botzner (@paginabianca)"
+options:
+  key:
+    description:
+      - Database key.
+    type: str
+    required: true
+  increment_int:
+    description:
+      - Integer amount to increment the key by.
+    required: false
+    type: int
+  increment_float:
+    description:
+      - Float amount to increment the key by.
+      - This only works with keys that contain float values
+        in their string representation.
+    type: str
+    required: false
+
+
+extends_documentation_fragment:
+  - community.general.redis.documentation
+
+seealso:
+    - module: community.general.redis_set
+    - module: community.general.redis_data_info
+    - module: community.general.redis
+'''
+
+EXAMPLES = '''
+- name: Increment integer key foo on localhost with no username and print new value
+  community.general.redis_incr:
+    login_host: localhost
+    login_password: supersecret
+    key: foo
+    increment_int: 1
+  register: result
+- name: Print new value
+  debug:
+    var: result.value
+
+- name: Increment float key foo by 20.4
+  community.general.redis_incr:
+    login_host: redishost
+    login_user: redisuser
+    login_password: somepass
+    key: foo
+    increment_float: '20.4'
+'''
+
+RETURN = '''
+value:
+  description: Incremented value of key
+  returned: on success
+  type: str
+  sample: '4039.4'
+msg:
+  description: A short message.
+  returned: always
+  type: str
+  sample: 'Incremented key: foo by 20.4 to 65.9'
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.general.plugins.module_utils.redis import (
+    fail_imports, redis_auth_argument_spec, RedisAnsible)
+
+
+def main():
+    redis_auth_args = redis_auth_argument_spec()
+    module_args = dict(
+        key=dict(type='str', required=True, no_log=False),
+        increment_int=dict(type='int', required=False),
+        increment_float=dict(type='str', required=False),
+    )
+    module_args.update(redis_auth_args)
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=False,
+        mutually_exclusive=[['increment_int', 'increment_float']],
+    )
+    fail_imports(module)
+
+    redis = RedisAnsible(module)
+    key = module.params['key']
+    increment_float = module.params['increment_float']
+    increment_int = module.params['increment_int']
+
+    result = {'changed': False}
+    if increment_float is not None:
+        increment = None
+        try:
+            increment = float(increment_float)
```

Comment:
Since you manually convert the value to float here, you should let Ansible do the conversion by specifying the type `float`.